File: server/EnglishTutorAI.Application/Models/CompletedDocumentListItem.cs
using EnglishTutorAI.Domain.Enums;

namespace EnglishTutorAI.Application.Models;

public class CompletedDocumentListItem
{
    public Guid DocumentId { get; set; }

    public required string Title { get; set; }

    public StudyTopic StudyTopic { get; set; }

    public required string Content { get; set; }

    public DateTime CompletedOn { get; set; }
}

File: client/src/app/dataModels/document/completedDocumentListItem.ts
import {StudyTopic} from "@/app/dataModels/enums/studyTopic.ts";

export interface CompletedDocumentListItem {
    documentId: string;
    title: string;
    studyTopic: keyof typeof StudyTopic;
    content: string;
    completedOn: string;
}